oshmem/mca/scoll/basic/scoll_basic_reduce.c

```diff
@@ -79,8 +79,9 @@ int mca_scoll_basic_reduce(struct oshmem_group_t *group,
         int i = 0;
 
         if (pSync) {
-            alg = (alg == SCOLL_DEFAULT_ALG ?
-                    mca_scoll_basic_param_reduce_algorithm : alg);
+            alg = (nlong ? (alg == SCOLL_DEFAULT_ALG ?
+                                   mca_scoll_basic_param_reduce_algorithm : alg) :
+                           SCOLL_ALG_REDUCE_CENTRAL_COUNTER );
```

Comment:
this is some kind of hack: in case if nlong == 0 then _algorithm_central_counter is called where is branch for such case - only barrier will be called